pkg/api/build.go

```diff
@@ -40,3 +41,29 @@ func (api Build) Jobs(c *gin.Context) {
 	}
 	c.JSON(http.StatusOK, jobs)
 }
+
+// Logs creates a new gin handler for the GET /job/:id/logs endpoint
+func (api Build) Logs(c *gin.Context) {
+	id := c.Params.ByName("id")
+	build, err := api.store.GetBuild(id)
+	if err != nil {
+		c.JSON(http.StatusNotFound, struct{}{})
+		return
+	}
+	if c.Query("stream") == "true" {
+		logReader, err := api.store.GetWorkerLogStream(build.Worker)
+		if err != nil {
+			c.JSON(http.StatusNotFound, struct{}{})
```

Comment:
Does an error here always indicate that `GetWorkerLogStream()` couldn't find the log stream?

Reply:
It means it could not fetch the log stream. We don't know all the errors that could be returned from k8s, though. All we have seen are missing logs due to clean ups